internal/models/repository.go

```diff
@@ -0,0 +1,14 @@
+package models
+
+type CommonRepository struct {
```

Comment:
Per the function call, should this be `RepositoryContext` or something?

Reply:
Yes, absolutely. :-)